QueryEngine/ResultSetIteration.cpp

```diff
@@ -2173,6 +2173,50 @@ bool ResultSetStorage::isEmptyEntryColumnar(const size_t entry_idx,
   return false;
 }
 
+namespace {
+
+template <typename T>
+inline size_t makeBinSearch(size_t l, size_t r, T&& is_empty_fn) {
```

Comment:
Since this is a free function, can you please change its name to be `make_bin_search` so that it matches our coding style guide?